datastore/src/main/java/com/openlattice/datastore/data/controllers/DataController.java

```diff
@@ -907,35 +903,65 @@ public Integer deleteEntityProperties(
     }
 
     @Timed
-    @RequestMapping(
-            path = { "/" + ENTITY_SET + "/" + SET_ID_PATH + "/" + NEIGHBORS },
-            method = RequestMethod.POST )
-    public Long deleteEntitiesAndNeighbors(
+    @Override
+    @DeleteMapping(
+            path = { "/" + ENTITY_SET + "/" + SET_ID_PATH + "/" + NEIGHBORS } )
+    public UUID deleteEntitiesAndNeighbors(
             @PathVariable( ENTITY_SET_ID ) UUID entitySetId,
             @RequestBody EntityNeighborsFilter filter,
-            @RequestParam( value = TYPE ) DeleteType deleteType ) {
+            @RequestParam( value = TYPE ) DeleteType deleteType,
+            @RequestParam( value = BLOCK, defaultValue = "true" ) boolean blockUntilCompletion ) {
         // Note: this function is only useful for deleting src/dst entities and their neighboring entities
         // (along with associations connected to all of them), not associations.
         // If called with an association entity set, it will simplify down to a basic delete call.
 
-        ensureEntitySetCanBeWritten( entitySetId );
+        // filter.entityKeyIds should be non-empty
+        Preconditions.checkArgument( !filter.getEntityKeyIds().isEmpty(), "EntityNeighborsFilter.entityKeyIds should be a non-empty set" );
+        
+        Set<UUID> dstEntitySetIds = filter.getDstEntitySetIds().orElse( Set.of() );
+        Set<UUID> srcEntitySetIds = filter.getSrcEntitySetIds().orElse( Set.of() );
+        Set<UUID> allEntitySetIds = Sets.union( srcEntitySetIds, dstEntitySetIds );
+        allEntitySetIds = Sets.union( allEntitySetIds, Set.of( entitySetId ) );
+        
+        ensureEntitySetsCanBeWritten( allEntitySetIds );
+
+        Map<UUID, Set<UUID>> entitySetIdEntityKeyIds = Maps.newHashMap();
+        entitySetIdEntityKeyIds.put( entitySetId, filter.getEntityKeyIds() );
+
+        // verify permission to delete
+        entitySetIdEntityKeyIds.forEach( ( key, val ) -> {
```

Comment:
This seems like both a bad idea and inefficient.

Reply:
So deeper explanation his will hit postgres with expensive queries several times. The problem is that none of the functions in the chain allow for batch parameters and the limiting step (an edge query to determine which entity sets are potential neighbor for auth checks) is expensive. Without digging into the query plan generated by postgres (most likely a sequential scan) my guess is that you'd get the same speed by rolling all of them up. In addition-- each call invokes an unnecessary query against postgres( Graph.checkForUnauthorized ). This isn't necessary because you already access checked all potential entity sets which would include any edges checked by checkForUnauthorized. We should do one or the other. Graph.checkForUnauthorized seems like the better approach since it limits edges to the edges affected by the delete and not all potential neighbor entity sets.